devtools/startup/DevToolsStartup.jsm

```diff
@@ -1423,58 +1423,21 @@ function recordReplayLog(text) {
 
 ChromeUtils.recordReplayLog = recordReplayLog;
 
-function getLoggedInUser() {
+function isLoggedIn() {
   const userPref = Services.prefs.getStringPref("devtools.recordreplay.user");
   if (userPref == "") {
     return;
   }
   const user = JSON.parse(userPref);
-  return user == "" ? null : user;
-}
-
-function saveRecordingInDB(description) {
-  let user;
-
-  if (isRunningTest()) {
-    user = {id: "77c6dc81-280d-4f28-971b-9915ab00f0f7"}
-  } else {
-    user = getLoggedInUser();
-  }
-
-  if (!user) {
-    return;
-  }
-
-  const pageUrl = Services.prefs.getStringPref(
-    "devtools.recordreplay.saveRecordingsUrl"
-  );
 
-  const body = {
-    user_id: user.id,
-    recording_id: description.recordingId,
-    id: description.recordingId,
-    url: description.url,
-    title: description.title,
-    duration: description.duration,
-    last_screen_data: description.lastScreenData || "",
-    last_screen_mime_type: description.lastScreenMimeType || "image/jpeg",
-  };
-
-  console.log(`>>> saving recording`, description.recordingId);
-  return fetch(`${pageUrl}/api/create-recording`, {
-    method: "post",
-    body: JSON.stringify(body),
-    headers: { "Content-Type": "application/json" },
-  })
-    .then(async (r) =>
-      console.log(`succeeded in creating recording`, await r.json())
-    )
-    .catch((err) => console.error(err));
+  // Older versions of Replay did not store the raw object from Auth0 and
+  // instead stored backend DB user info, so we need to explicitly look for
+  // "sub", not just any parsed object.
+  return user == "" ? null : !!user?.sub;
```

Comment:
Should this return a bool in all cases? Right not it looks like it could sometimes return null.

Reply:
Ah good call, I tried to make it consistent and missed the `null`!